client/landing/jetpack-cloud/components/sidebar/index.jsx

```diff
@@ -0,0 +1,208 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import Sidebar from 'layout/sidebar';
+import SidebarFooter from 'layout/sidebar/footer';
+import SidebarItem from 'layout/sidebar/item';
+import SidebarMenu from 'layout/sidebar/menu';
+import SidebarRegion from 'layout/sidebar/region';
+import ExpandableSidebarMenu from 'layout/sidebar/expandable';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+class JetpackCloudSidebar extends Component {
+	static propTypes = {
+		context: PropTypes.shape( {
+			path: PropTypes.string,
+		} ),
+	};
+
+	state = {
+		expandedSections: [],
+	};
+
+	componentDidMount() {
+		const section = this.getSectionFromCurrentPath();
+
+		if ( section ) {
+			this.toggleSection( section );
+		}
+	}
+
+	/**
+	 * Get current section name from path. When there's no section, we're in the dashboard.
+	 *
+	 * @returns {string} Current section
+	 */
+	getSectionFromCurrentPath() {
+		// If no section is present in the path, return 'dashboard'
+		return this.props.context.path.split( '/' )[ 2 ] ?? 'dashboard';
+	}
+
+	/**
+	 * Get current subsection name from path.
+	 *
+	 * @returns {string|null} Current subsection or null
+	 */
+	getSubsectionFromCurrentPath() {
+		return this.props.context.path.split( '/' )[ 3 ] ?? null;
+	}
+
+	/**
+	 * Toggle an expandable section.
+	 *
+	 * @param   {string} section Section to be toggled
+	 * @returns {void}
+	 */
+	toggleSection( section ) {
+		if ( this.isExpanded( section ) ) {
+			this.setState( {
+				expandedSections: this.state.expandedSections.filter( item => item !== section ),
+			} );
+		} else {
+			this.setState( {
+				expandedSections: [ ...this.state.expandedSections, section ],
+			} );
+		}
+	}
+
+	/**
+	 * Check if a section menu is expanded.
+	 *
+	 * @param   {string} section Section name
+	 * @returns {boolean}        Is the section expanded
+	 */
+	isExpanded( section ) {
+		return this.state.expandedSections.includes( section );
+	}
+
+	/**
+	 * Check if a menu item is selected based on the current path.
+	 *
+	 * @param   {string}  path Menu item path
+	 * @returns {boolean}      Is the menu item selected
+	 */
+	isSelected( path ) {
+		const section = this.getSectionFromCurrentPath();
+		const subsection = this.getSubsectionFromCurrentPath();
+
+		if ( ! subsection ) {
+			return section === path;
+		}
+
+		return section.concat( '/', subsection ) === path;
+	}
+
+	handleExpandableMenuClick( section ) {
```

Comment:
I don't see a good reason to have this indirection. Seems like instead of `handleExpandableMenuClick` we could directly use `toggleSection`?

Reply:
We're using `toggleSection` method in `componentDidMount` for setting up an initial UI state.

If we wanted to use `handleExpandableMenuClick` directly there, we would end up with something that doesn't feel good to me:

```js
this.handleExpandableMenuClick( section )();
```

What do you think?

Reply:
Makes sense, let's keep it then 👍